support-metrics-client/src/main/java/io/confluent/support/metrics/SupportedServerStartable.java

```diff
@@ -126,9 +123,12 @@ public void startup() {
 
   public void shutdown() {
     try {
-      if (metricsThread != null) {
-        metricsThread.interrupt();
-        metricsThread.join();
+      log.info("Shutting down SupportedServerStartable");
+      if (metricsReporter != null) {
+        metricsReporter.close();
+        log.info("Waiting for metrics thread to exit");
+        metricsReporter.join();
```

Comment:
should we wait for a limited time here?

Reply:
I am not sure if it could affect anything upstream if JVM shuts down in the middle of sending metrics. We can consider doing this in the future, if this continues being a problem. What do you think?

Reply:
Are there are any hard reasons to guarantee metrics upload? I thought it's ok to miss out on metrics which were collected in the last interval before shutdown.

Also, if we really want to guarantee metrics upload then maybe we don't want this to be a daemon, as the join does defeat the purpose of having a background worker thread which we don't care about stopping/killing when the JVM goes down.

Reply:
Yeah, I agree. I don't have enough context about why this code was written the way it is. I would have assumed that if a thread is declared daemon then we would not even bother attempting a graceful shutdown.

In my opinion, we should either just not care about the metrics thread (i.e. do not signal it to close and do not wait for it to exit) or we should wait for it to stop gracefully. I am not sure why we'd do a timed wait in this context.

Being a bit conservative, I think keeping the graceful shutdown behavior is safest for now.

Reply:
The only concern I have is that the `join` waits for a very long time. If you think that is not not possible, then let's go ahead and merge this.